### DISCUSSION TITLE
Using Quarto with Julia-1.8.5

### DISCUSSION THREAD

**Discussion by ProfCelia** [quarto-dev/quarto-cli#4625] [Q&A]
Previously I was able to use Quarto with earlier versions of Julia at least 1.7, Now Quarto cannot find Julia-1.8.5.

I have tried everything I can see suggested on the internet to no avail. Is this simply my problem or does Quarto plan to address it sometime.

**Comment by cscheid** [comment]
We haven't changed anything about our Julia setup, so unfortunately we can't address that.

If I had to guess, I'd say something has changed with your Python environment and now Jupyter can't find the Julia engine. You might need to reinstall IJulia.

**Reply by ProfCelia** [reply]
Jupyter has no problem finding my julia engine. Jupyter notebooks work just fine. It's quarto that doesn't work. I have rebuilt IJulia multiple times I tried startstarting the notebook in a different directory, I've dried using the pre-release version of Quarto. I guess I just have to wait around until things eventually get fixed. I am not going to subject my students to this nonsense. Quarto is a lovely product, too bad it doesn't work for me.

**Reply by cscheid** [reply]
I'm sorry this setup is not working for you!

> Jupyter has no problem finding my julia engine. Jupyter notebooks work just fine. 

What I suspect could be happening is that you have more than one version of Python installed, and different Jupyter installations on each. If the shell environment you get from quarto doesn't match the one you get from jupyter notebook, then things might not work out. You can try to compare the versions you get with `quarto check`, which will list the Jupyter engines that quarto finds.

You can also try running the following Python snippet in a `python` code cell and in a jupyter notebook, and inspect the difference:

````
```{python}
import sys
print(sys.path)
print(sys.version_info)
```
````

The reason that the _Python_ version matters is that if you have different installed versions of Python (or different virtual environments), then Jupyter engines will store information about themselves in different places. That will ultimately affect IJulia.

> I am not going to subject my students to this nonsense.

I totally understand the frustration and reservations: I was [a professor](https://scholar.google.com/citations?user=8uHMJjsAAAAJ&hl=en) for the better part of a decade! If you choose to go down the route of diagnosing these Jupyter issues, let us know. There's a large group of expert Python and Julia folks in the forums here as well who might be able to help.

**Reply by kdheepak** [reply]
fwiw I'm using the latest stable Julia release (1.8.5) and a Quarto pre-release, and everything seems to be working fine for me.

**Reply by fortunewalla** [reply]
> Jupyter has no problem finding my julia engine. Jupyter notebooks work just fine. It's quarto that doesn't work. I have rebuilt IJulia multiple times I tried startstarting the notebook in a different directory, I've dried using the pre-release version of Quarto. I guess I just have to wait around until things eventually get fixed. I am not going to subject my students to this nonsense. Quarto is a lovely product, too bad it doesn't work for me.

As the above person mentioned, can you post the results of the `quarto check` command in your system? and also the error messages or steps taken?

I too had many issues but for me it was mainly about entering the path in the right way.

### 1. I created the following sample file `qdoc.qmd` to test out Julia using Quarto through Jupyter kernel.

```md
---
title: "Demo to test Julia code in Quarto"
author: "Fortune Walla"
date: "3/5/2023"
format:
  html:
    code-fold: true
jupyter: julia-1.8
---

## Julia code below.

```{julia}

println(1+1)

println("Hello from Julia land through Quarto")

VERSION

    ```

```


### 2. I tested with HTML render

```
quarto render qdoc.qmd --to html

Executing 'qdoc.ipynb'
  Cell 1/1...Done

pandoc
  to: html
  output-file: qdoc.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png

metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: Demo to test Julia code in Quarto
  author: Fortune Walla
  date: 3/5/2023
  jupyter: julia-1.8

Output created: qdoc.html
```

### 3. Screenshot of HTML output

![image](https://user-images.githubusercontent.com/7158596/222975085-cca9f181-74b6-46ef-aee0-9f281fe1eb0e.png)

### 4. Similar step to render to PDF

```
quarto render qdoc.qmd --to pdf

Executing 'qdoc.ipynb'
  Cell 1/1...Done

pandoc
  to: latex
  output-file: qdoc.tex
  standalone: true
  pdf-engine: xelatex
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf

metadata
  documentclass: scrartcl
  classoption:
    - DIV=11
    - numbers=noendperiod
  papersize: letter
  header-includes:
    - '\KOMAoption{captions}{tableheading}'
  block-headings: true
  title: Demo to test Julia code in Quarto
  author: Julia Quarto
  date: 3/5/2023
  jupyter: julia-1.8

running xelatex - 1
  This is XeTeX, Version 3.141592653-2.6-0.999994 (TeX Live 2022) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode

running xelatex - 2
  This is XeTeX, Version 3.141592653-2.6-0.999994 (TeX Live 2022) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode


Output created: qdoc.pdf
```

### 5. Screenshot of PDF output.

![image](https://user-images.githubusercontent.com/7158596/222975172-1e9c1db8-5144-467a-8b00-2f7145a02dc9.png)

**Comment by DhruvaSambrani** [comment]
~~https://github.com/quarto-dev/quarto-cli/issues/4821 Fixes this~~
See below

**Reply by cscheid** [reply]
For everyone else, I want to be a little more precise. #4821 specifically fixes the issue in the case that you're using IJulia's embedded miniconda setup. If you're not, then you're not affected by this bug (nor will it fix it!) I just want to make sure we keep accurate information, since debugging these issues is already hard enough by itself.

**Reply by kdheepak** [reply]
That clarification is useful! I think it’ll be useful to add some documentation on how things were in quarto (how does quarto know which jupyter instance to use etc) and maybe provide some recommendations for how to set it up for Julia users. 

Recently I had an error occur because quarto was picking up my homebrew python installation on my Mac and failing because of jupyter not being installed but only when I tried to run using the render button from VSCode. This worked fine when I ran it using the terminal (inside and outside VSCode). I found that extremely surprising, and solved it by just uninstalling my homebrew python installation.

I think more guidance on recommendations for how to set it up for Julia users (using both the built in miniconda install and a explicit miniconda install) would be great to have as part of the official documentation.
